driver/bpf/plumbing_helpers.h

```diff
@@ -423,93 +423,30 @@ static __always_inline bool drop_event(void *ctx,
 				       struct scap_bpf_settings *settings,
 				       enum syscall_flags drop_flags)
 {
-	if (!settings->dropping_mode)
+	if (drop_flags & UF_NEVER_DROP)
```

Comment:
While logically equals, ported to follow same kmod if-else logic.

Reply:
> we manage this logic in userspace

This would involve two syscalls per second, at very precise timing, when we're running under heavy load already (otherwise we don't need to enter sampling mode).

IMO, we could handle this with two sets of interesting syscalls: the "default" one and the "sampling" one; we'd flip between them periodically in the kernel.

> why we need to use these 2 `PPME_DROP_E` and `PPME_DROP_X` events

The commercial sysdig agent uses these to ensure any metrics are properly scaled even when sampling.

Reply:
> IMO, we could handle this with two sets of interesting syscalls: the "default" one and the "sampling" one; we'd flip between them periodically in the kernel.

That's exactly our idea, find a set that could be ok for the sampling and enable it when necessary

Reply:
> That's exactly our idea, find a set that could be ok for the sampling and enable it when necessary

Yeah, but I believe we'd still need to flip the sets in the kernel, not userspace, due to the timing (we need <1ms accuracy for sr=128 and that's precisely when we consider ourselves to be overloaded and so can't really rely on such precise timing in userspace).

This is probably all solvable somehow, but I'm afraid not in the context of a cleanup; we'd need to do some serious design work first I'm afraid.

Reply:
> This is probably all solvable somehow, but I'm afraid not in the context of a cleanup; we'd need to do some serious design work first I'm afraid.

Of course, this is something we should leave for a feature PR.